src/Facility.Core/ServiceErrorDto.cs

```diff
@@ -37,4 +37,15 @@ public ServiceErrorDto(string? code, string? message)
 		get => DetailsObject?.AsJObject();
 		set => DetailsObject = ServiceObject.Create(value);
 	}
+
+	/// <summary>
+	/// Returns the error details as an instance of the specified DTO.
+	/// </summary>
+	public T? GetDetailsDto<T>()
+		where T : ServiceDto => DetailsObject is { } obj ? obj.ToDto<T>() : default;
+
+	/// <summary>
+	/// Sets the details from the specified DTO.
+	/// </summary>
+	public void SetDetailsDto(ServiceDto? obj) => DetailsObject = ServiceObject.FromDto(obj);
```

Comment:
They make it more discoverable. Without these, I'd probably add a `GetDetails<T>()` extension method in my own code (Dto suffix seems verbose), but probably not the setter because I'd be more likely to use initializer syntax.

Reply:
I wasn't sure about the `Dto` suffix, but you're probably right.

Callers of the [helper methods](https://github.com/FacilityApi/FacilityCSharp/blob/2d8cb91089eb071891f026486c24e4d4b59e2a05/src/Facility.Core/ServiceErrors.g.cs#L78) might still use `SetDetails`, and the symmetry is appropriate, so I'd probably keep it, but you're right that it's not useful with initializer syntax.

Reply:
A fluent-style `ServiceErrorDto WithDetails<T>(T dto)` method (or extension method) could also make initialization easier.